official/vision/beta/projects/yolo/tasks/yolo.py

```diff
@@ -0,0 +1,437 @@
+ 
+    Args:
+      inputs: a dictionary of input tensors.
+      model: the keras.Model.
+      metrics: a nested structure of metrics objects.
+      
+    Returns:
+      A dictionary of logs.
+    """
+    image, label = inputs
+
+    # Step the model once
+    y_pred = model(image, training=False)
+    y_pred = tf.nest.map_structure(lambda x: tf.cast(x, tf.float32), y_pred)
+    (_, metric_loss, loss_metrics) = self.build_losses(y_pred['raw_output'],
+                                                       label)
+    logs = {self.loss: metric_loss}
+
+    # Reorganize and rescale the boxes
+    boxes = self._reorg_boxes(y_pred['bbox'], y_pred['num_detections'], image)
+    label['groundtruths']["boxes"] = self._reorg_boxes(
+        label['groundtruths']["boxes"], label['groundtruths']["num_detections"],
+        image)
+
+    # Build the input for the coc evaluation metric
+    coco_model_outputs = {
+        'detection_boxes': boxes,
+        'detection_scores': y_pred['confidence'],
+        'detection_classes': y_pred['classes'],
+        'num_detections': y_pred['num_detections'],
+        'source_id': label['groundtruths']['source_id'],
+        'image_info': label['groundtruths']['image_info']
+    }
+
+    # Compute all metrics
+    if metrics:
+      logs.update(
+          {self.coco_metric.name: (label['groundtruths'], coco_model_outputs)})
+      for m in metrics:
+        m.update_state(loss_metrics[m.name])
+        logs.update({m.name: m.result()})
+    return logs
+
+  def aggregate_logs(self, state=None, step_outputs=None):
+    """Get Metric Results."""
+    if not state:
+      self.coco_metric.reset_states()
+      state = self.coco_metric
+    self.coco_metric.update_state(step_outputs[self.coco_metric.name][0],
+                                  step_outputs[self.coco_metric.name][1])
+    return state
+
+  def reduce_aggregated_logs(self, aggregated_logs, global_step=None):
+    """Reduce logs and remove unneeded items. Update with COCO results."""
+    res = self.coco_metric.result()
+    return res
+
+  def initialize(self, model: tf.keras.Model):
+    """Loading pretrained checkpoint."""
+
+    if not self.task_config.init_checkpoint:
+      logging.info("Training from Scratch.")
+      return
+
+    ckpt_dir_or_file = self.task_config.init_checkpoint
+    if tf.io.gfile.isdir(ckpt_dir_or_file):
+      ckpt_dir_or_file = tf.train.latest_checkpoint(ckpt_dir_or_file)
+
+    # Restoring checkpoint.
+    if self.task_config.init_checkpoint_modules == 'all':
+      ckpt = tf.train.Checkpoint(model=model)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.expect_partial().assert_existing_objects_matched()
+    elif self.task_config.init_checkpoint_modules == 'backbone':
+      ckpt = tf.train.Checkpoint(backbone=model.backbone)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.expect_partial().assert_existing_objects_matched()
+    elif self.task_config.init_checkpoint_modules == 'decoder':
```

Comment:
We usually have this as a list of objects to load. ['backbone', 'decoder'] if you want to load both. Can you look at retinanet, or semantic segmentation?

Reply:
done